gcp_variant_transforms/transforms/merge_headers_test.py

```diff
@@ -102,19 +103,110 @@ def test_combine_multiple_headers_as_accumulators(self):
     self.assertItemsEqual(merged_headers.infos.keys(), ['NS', 'AF', 'NS2'])
     self.assertItemsEqual(merged_headers.formats.keys(), ['GT', 'GQ', 'GQ2'])
 
+  def test_combine_two_conflicting_headers(self):
+    # These two headers have type conflict (Integer vs Float).
+    lines_1 = [
+      '##fileformat=VCFv4.2\n',
+      '##INFO=<ID=NS,Number=1,Type=Integer,Description="Number samples">\n',
+      '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample1 Sample2\n']
+    lines_2 = [
+      '##fileformat=VCFv4.2\n',
+      '##INFO=<ID=NS,Number=1,Type=Float,Description="Number samples">\n',
+      '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample3\n']
+
+    vcf_reader_1 = vcf.Reader(fsock=iter(lines_1))
+    vcf_reader_2 = vcf.Reader(fsock=iter(lines_2))
+    headers_1 = self._get_header_from_reader(vcf_reader_1)
+    headers_2 = self._get_header_from_reader(vcf_reader_2)
+
+    combiner_fn = merge_headers._MergeHeadersFn(force_merge_conflicts=False)
+
+    try:
+      merged_headers = combiner_fn.create_accumulator()
+      merged_headers = combiner_fn.add_input(merged_headers, headers_1)
+      merged_headers = combiner_fn.add_input(merged_headers, headers_2)
+      merged_headers = combiner_fn.extract_output(merged_headers)
+      self.fail('Incompatible VCF headers must throw an exception.')
+    except ValueError:
+      pass
```

Comment:
instead of doing the try except for value error, you can write `with self.assertRaises(ValueError):` and then have the rest of the calls nested within there.

Reply:
Done!